### ISSUE TITLE
`azurerm_subnet` not deleted while update `azurerm_virtual_network` at the same time

### ISSUE THREAD

**Issue by teowa** [hashicorp/terraform-provider-azurerm#21354]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.51.0

### Affected Resource(s)/Data Source(s)

azurerm_virtual_network, azurerm_subnet

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "wt-example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "example-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "example" {
  name                 = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.0/24"]
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: A resource with the ID "/subscriptions/xxx/resourceGroups/wt-example-resources/providers/Microsoft.Network/virtualNetworks/example-vnet/subnets/example-subnet" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_subnet" for more information.
│
│   with azurerm_subnet.example,
│   on main.tf line 21, in resource "azurerm_subnet" "example":
│   21: resource "azurerm_subnet" "example" {
```

### Expected Behaviour

subnet `example-subnet` successfully deleted after step 2 

### Actual Behaviour

subnet `example-subnet` still exist after step 2 and cannot recreate at step 3.

### Steps to Reproduce

1. run `terraform apply` with above example config

2. remove subnet and update vnet (add a tag) at same time, run `terraform apply`, but found the subnet is not removed
```hcl
provider "azurerm" {
  features {}
}
data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "example" {
  name     = "wt-example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "example-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  tags = {
    env = "test"
  }
}

```

3: recreate the subnet with step1 config

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/21348
https://github.com/hashicorp/terraform-provider-azurerm/pull/21352

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
